source/manuals/programming-languages/nodejs/index.html.md.erb

```diff
@@ -82,24 +116,339 @@ Keeping tests in the same directory makes them easier to find and more obvious w
+API. So the application should handle all operational errors and recover from
+them.
+
+## Node.js's HTTP server
+
+> Offload Node's server as much as possible
+
+Don't expose Node's HTTP server to the public. Use a reverse proxy to serve
+static assets, and cache content as much as possible. Implement adequate
+supervising: [pm2](http://pm2.keymetrics.io/) is recommended for Node.js-specific
+servers.
+
+## Transpiling
+
+> Stick to JavaScript
+
+Avoid anything that compiles to JavaScript (except for static type checking, see
+below). Examples to avoid include CoffeeScript, PureScript and [many others](https://github.com/jashkenas/coffeescript/wiki/List-of-languages-that-compile-to-JS).
+
+If you would like to use static typing consider using JavaScript extensions like
+[TypeScript](http://www.typescriptlang.org/), or [Flow](https://flow.org/). This
+has the advantage of making it easier to prevent runtime errors, by adding type
+information to variables or parameters and transpiling the source to JavaScript,
+reporting errors when wrong types are used.
+
+However be aware that there are disadvantages too: augmented source code will no
+longer be executable uncompiled. You may also need some extra type definitions if you
+want to use external libraries.
+
+Generally, anything that departs from the standard JavaScript syntax in a way
+that a Node developer would have trouble reading your code is advised against.
+
+## Frameworks
+
+> Use Express for web applications, avoid lesser-known frameworks
+
+Express is very common for writing web applications that talk to back-end
+APIs. If you find yourself looking for an <abbr
+title="model-view-controller">MVC</abbr> framework or if you need an <abbr
+title="object relationship manager">ORM</abbr> to manage records in a database,
+[you probably shouldn't be using Node.js in the first
+place](/standards/programming-languages.html#javascript).
+
+
+## Libraries
+
+> Avoid libraries that produce esoteric code, or that have a steep learning curve
+
+Using advanced libraries can make code very hard to read for developers not familiar
+with them, as useful as they may be. For instance, [Ramda](http://ramdajs.com/)
+lets you write:
+
+```js
+R.cond([
+ [R.is(Number), R.identity],
+ [R.is(String), parseInt],
+ [R.T, R.always(NaN)]
+])
+```
+
+which is compact and useful, but not easily understood up by a developer not
+familiar with it.
+
+Generally, readable code is better than compact or advanced code. Optimisation
+can lead to very arcane code, so should only be used when necessary.
+
+
+```js
+// Prefer:
+for (let i = 0; i < 10; i += 1) {
+  for (let j = 0; j < 10; j += 1) {
+    console.log(i, j)
+  }
+}
+
+// Over:
+for (let i=0,j=0;i<10 && j<10;j++,i=(j==10)?i+1:i,j=(j==10)?j=0:j,console.log(i,j)){}
+```
+
+
+## Node Package Manager (<abbr title="Node package manager">NPM</abbr>)
+
+> Limit your reliance on external code, and check the NPM packages you use
```

Comment:
I don't think this is good advice. The power of Node is that there is NPM behind it. IMHO I think the advice should be the opposite.
``` Don't re-invent the wheel, the NPM registry has packages to help solve a lot of common problems. Leverage existing open-source code where possible to maximise your productivity```

Followed by some warnings and advice on how to judge modules prior to using them.

Reply:
Although I agree with `Don't re-invent the wheel` ideology, I'm not a big fan of importing big library just to use one small function within it.

Reply:
I don't disagree, but I've only found this a significant issue on the client rather than server side. The vast majority of NPM libraries in my experience are pretty small (<3 files). Notable exceptions  include `lodash`, but that can be imported at a granular level i.e: `lodash/get`.

Reply:
Sounds fair.

Reply:
I'll change that to general advice that's less opinionated.